src/generate.ts

```diff
@@ -22,5 +28,101 @@ export function generate(specQ: SpecQuery, schema: Schema, opt: QueryConfig = DE
     }
   });
 
+  let ENCODING_QUERY_INDEX: Dict<EncodingQuery> = {};
```

Comment:
Extract methods for all these.  

In generate, you want to keep method short, keep things in the same method have same level of details. 

The upper part is very high level code that calls other methods.  So it's bad that you add low level code just right here.   This way it's hard to to targeted unit test.